llvm/include/llvm/ProfileData/InstrProf.h

```diff
@@ -449,6 +449,19 @@ class InstrProfSymtab {
     return "** External Symbol **";
   }
 
+  // Returns the canonial name of the given PGOName. In a canonical name, all
+  // suffixes that begins with "." except ".__uniq." are stripped.
+  // FIXME: Unify this with `FunctionSamples::getCanonicalFnName`. And both
+  // should probably preserve `.specialized.<NSpecs>` suffix now that
```

Comment:
It isn't clear to me why we would want to preserve .specialized, since presumably those specialized copies are created after PGO annotation / matching.

Reply:
ah you are correct. I initially thought stripping `.specialized.<NSpecs>` from function names gives in-accurate profile matching (maps names corresponding to different functions to one `&F`), but the specialized copies doesn't exist at profile annotation. I'm not aware of other places where the profiles for specialized copies (if the specialized copies exist in the instrumented or sampled binary) are handled, but I'll remove the comment.